File: ios/Releaf/Features/Onboarding/OnboardingTokens.swift
/*
 * OnboardingTokens.swift
 *
 * Literal colour/typography tokens extracted from
 * docs/onboarding/source/onboarding.css. The wizard intentionally
 * uses fixed coral values (rather than the app's themeable accent)
 * so the first-run experience has a consistent brand identity.
 */

import SwiftUI

enum OnboardTokens {
    // Colours (from onboarding.css)
    static let coral           = Color(hex: 0xFF5F4E)
    static let coralFaded      = Color(hex: 0xFFB8B2)
    static let coralSoftBg     = Color(hex: 0xFFF5F4)
    static let coralSoftBorder = Color(hex: 0xFFD5D0)
    static let modalBg         = Color(hex: 0xFDF8F4)
    static let cardBg          = Color.white
    static let textPrimary     = Color(hex: 0x1B1B1D)
    static let textBody        = Color(hex: 0x4A4845)
    static let textMuted       = Color(hex: 0x7A7670)
    static let textSubtle      = Color(hex: 0x9E9990)
    static let borderRest      = Color(hex: 0xE2DBD4)
    static let lineFill        = Color(hex: 0xEDE8E3)
    static let ghostHoverBg    = Color(hex: 0xF0ECE8)
    static let todoTaskBg      = Color(hex: 0xFFF0E8)
    static let todoTaskFg      = Color(hex: 0xE06020)
    static let scanDoneBg      = Color(hex: 0xFFD5D1)
    static let scanDoneFg      = Color(hex: 0xC0392B)
    static let photoFrameBg    = Color(hex: 0xF0ECE8)
    static let photoFrameBorder = Color(hex: 0xE0D8D0)

    // App-icon gradient (see docs/onboarding/source/app-icon.svg)
    static let iconGradientStart = Color(hex: 0x123524)
    static let iconGradientEnd   = Color(hex: 0x3F7D58)
    static let iconSurface       = Color(hex: 0xF7F5EF)
    static let iconLine          = Color(hex: 0x2F5D50)
    static let iconDotFill       = Color(hex: 0xD98324)

    // Typography
    static let headline      = Font.system(size: 23, weight: .heavy, design: .default)
    static let body          = Font.system(size: 15, weight: .regular, design: .default)
    static let badge         = Font.system(size: 11, weight: .bold, design: .default)
    static let button        = Font.system(size: 14, weight: .bold, design: .default)
    static let skip          = Font.system(size: 12, weight: .regular, design: .default)
    static let calendarHeader = Font.system(size: 11, weight: .bold, design: .default)
    static let calendarNumber = Font.system(size: 38, weight: .heavy, design: .default)
    static let ctaLabel      = Font.system(size: 13, weight: .regular, design: .default)
    static let ctaLabelBold  = Font.system(size: 13, weight: .bold,    design: .default)
    static let scanPill      = Font.system(size: 10, weight: .semibold, design: .default)
    static let todoItem      = Font.system(size: 13, weight: .regular, design: .default)
    static let photoBadge    = Font.system(size: 12, weight: .semibold, design: .default)
}

File: android/app/src/main/java/app/releaf/mobile/features/onboarding/OnboardingTokens.kt
/*
 * OnboardingTokens.kt
 *
 * Literal colour/typography tokens extracted from
 * docs/onboarding/source/onboarding.css. The wizard intentionally
 * uses fixed coral values (rather than the themeable [AppAccent]) so
 * the first-run experience has a consistent brand identity regardless
 * of the active accent palette.
 */

package app.releaf.mobile.features.onboarding

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

internal object OnboardTokens {
    // Colours (from onboarding.css)
    val Coral          = Color(0xFFFF5F4E)
    val CoralFaded     = Color(0xFFFFB8B2)
    val CoralSoftBg    = Color(0xFFFFF5F4)
    val CoralSoftBorder= Color(0xFFFFD5D0)
    val ModalBg        = Color(0xFFFDF8F4)
    val CardBg         = Color(0xFFFFFFFF)
    val TextPrimary    = Color(0xFF1B1B1D)
    val TextBody       = Color(0xFF4A4845)
    val TextMuted      = Color(0xFF7A7670)
    val TextSubtle     = Color(0xFF9E9990)
    val BorderRest     = Color(0xFFE2DBD4)
    val LineFill       = Color(0xFFEDE8E3)
    val GhostHoverBg   = Color(0xFFF0ECE8)
    val TodoTaskBg     = Color(0xFFFFF0E8)
    val TodoTaskFg     = Color(0xFFE06020)
    val ScanDoneBg     = Color(0xFFFFD5D1)
    val ScanDoneFg     = Color(0xFFC0392B)
    val PhotoFrameBg   = Color(0xFFF0ECE8)
    val PhotoFrameBorder = Color(0xFFE0D8D0)

    // Gradient stops for the Releaf app icon.
    val IconGradientStart = Color(0xFF123524)
    val IconGradientEnd   = Color(0xFF3F7D58)
    val IconSurface       = Color(0xFFF7F5EF)
    val IconLine          = Color(0xFF2F5D50)
    val IconDotFill       = Color(0xFFD98324)

    // Typography
    val Headline = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.ExtraBold,
        fontSize   = 23.sp,
        lineHeight = 29.sp,
    )
    val Body = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Normal,
        fontSize   = 15.sp,
        lineHeight = 24.sp,
    )
    val Badge = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Bold,
        fontSize   = 11.sp,
        letterSpacing = 0.6.sp,
    )
    val Button = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Bold,
        fontSize   = 14.sp,
    )
    val Skip = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Normal,
        fontSize   = 12.sp,
    )
    val CalendarHeader = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Bold,
        fontSize   = 11.sp,
        letterSpacing = 0.5.sp,
    )
    val CalendarNumber = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.ExtraBold,
        fontSize   = 38.sp,
        lineHeight = 38.sp,
    )
    val CtaLabel = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Normal,
        fontSize   = 13.sp,
        lineHeight = 18.sp,
    )
    val ScanPill = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.SemiBold,
        fontSize   = 10.sp,
    )
    val TodoItem = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Normal,
        fontSize   = 13.sp,
    )
    val PhotoBadge = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.SemiBold,
        fontSize   = 12.sp,
    )
}
